pkg/common/resource.go

```diff
@@ -244,3 +268,35 @@ func Add(left *si.Resource, right *si.Resource) *si.Resource {
 	}
 	return result
 }
+
+func Sub(left *si.Resource, right *si.Resource) *si.Resource {
+	if left == nil {
+		left = &si.Resource{}
+	}
+	if right == nil {
+		return &si.Resource{}
+	}
+
+	// neither are nil, clone one and sub the other
+	rb := NewResourceBuilder()
+	for k, v := range left.Resources {
+		if rightValue, ok := right.Resources[k]; ok {
+			rb.AddResource(k, v.Value - rightValue.Value)
+		} else {
+			rb.AddResource(k, v.Value)
+		}
+	}
+	return rb.Build()
```

Comment:
if *right* contains resources *left* does not have we do not subtract them.
This is contrary to what we do in add where we properly merge the two.

We should behave the same way in add and sub